helpers/portal.go

```diff
@@ -239,12 +241,45 @@ func RestartDashboard() error {
 		return err
 	}
 
-	deploymentsClient := clientset.AppsV1().Deployments(data.AppConfig.TykPodNamespace)
+	if data.AppConfig.DashboardDeploymentName == "" {
+		ls := metav1.LabelSelector{MatchLabels: map[string]string{
+			constants.TykBootstrapLabel: constants.TykBootstrapDashboardDeployLabel,
+		}}
+
+		if data.AppConfig.ReleaseName != "" {
+			ls.MatchLabels[constants.TykBootstrapReleaseLabel] = data.AppConfig.ReleaseName
+		}
+
+		deployments, err := clientset.
+			AppsV1().
+			Deployments(data.AppConfig.TykPodNamespace).
+			List(
+				context.TODO(),
+				metav1.ListOptions{
+					LabelSelector: labels.Set(ls.MatchLabels).String(),
+				},
+			)
+		if err != nil {
+			return errors.New(fmt.Sprintf("failed to list Tyk Dashboard Deployment, err: %v", err))
+		}
+
+		for _, deployment := range deployments.Items {
+			data.AppConfig.DashboardDeploymentName = deployment.ObjectMeta.Name
```

Comment:
Should we throw an error if we found multiple dashboard deployments?
Otherwise we will bootstrap any random bootstrap deployment.

Reply:
that's a good point. correct me if i am wrong but we only need dashboard deployment name to restart the dashboard. if that's the case, do you think it'll be a problem? I mean, the code piece restarts Dashboard deployment without relying on the number of pods controlled by this Deployment.

Reply:
If we expect to have just one deployment, do we need a for loop?

Reply:
@komalsukhani I have made the necessary updates, including adjustments related to discovering service ports. Could you please review it again?